src/sage/graphs/graph.py

```diff
@@ -5069,6 +5069,202 @@ def minor(self, H, solver=None, verbose=0, *, integrality_tolerance=1e-3):
+        p = MixedIntegerLinearProgram(solver=solver)
+
+        # We use frozenset((u, v)) to avoid confusion between (u, v) and (v, u)
+
+        # rs = Representative set of a vertex
+        # for h in H, v in G is such that rs[h,v] == 1 if and only if v
+        # is a representative of h in self
+        rs = p.new_variable(binary=True)
+
+        for v in self:
+            p.add_constraint(p.sum(rs[h, v] for h in H), max=1)
+
+        # We ensure that the set of representatives of a
+        # vertex h contains a tree, and thus is connected
+
+        # edges represents the edges of the tree
+        edges = p.new_variable(binary=True)
+
+        # there can be a edge for h between two vertices
+        # only if those vertices represent h
+        for u, v in self.edge_iterator(labels=None):
+            fuv = frozenset((u, v))
+            for h in H:
+                p.add_constraint(edges[h, fuv] - rs[h, u], max=0)
+                p.add_constraint(edges[h, fuv] - rs[h, v], max=0)
+
+        # The number of edges of the tree in h is exactly the cardinal
+        # of its representative set minus 1
+
+        for h in H:
+            p.add_constraint(p.sum(edges[h, frozenset(e)] for e in self.edge_iterator(labels=None))
+                             - p.sum(rs[h, v] for v in self), min=-1, max=-1)
+
+        # a tree  has no cycle
+        epsilon = 1/(5*Integer(self.order()))
+        r_edges = p.new_variable(nonnegative=True)
+
+        for h in H:
+            for u, v in self.edge_iterator(labels=None):
+                p.add_constraint(r_edges[h, (u, v)] + r_edges[h, (v, u)] - edges[h, frozenset((u, v))], min=0)
+
+            for v in self:
+                p.add_constraint(p.sum(r_edges[h, (u, v)] for u in self.neighbor_iterator(v)), max=1-epsilon)
+
+        # Once the representative sets are described, we must ensure
+        # there are arcs corresponding to those of H between them
+        h_edges = p.new_variable(nonnegative=True)
+
+        for h1, h2 in H.edge_iterator(labels=None):
+
+            for v1, v2 in self.edge_iterator(labels=None):
+                fv1v2 = frozenset((v1, v2))
+                p.add_constraint(h_edges[(h1, h2), fv1v2] - rs[h2, v2], max=0)
+                p.add_constraint(h_edges[(h1, h2), fv1v2] - rs[h1, v1], max=0)
+
+                p.add_constraint(h_edges[(h2, h1), fv1v2] - rs[h1, v2], max=0)
+                p.add_constraint(h_edges[(h2, h1), fv1v2] - rs[h2, v1], max=0)
+
+            p.add_constraint(p.sum(h_edges[(h1, h2), frozenset(e)] + h_edges[(h2, h1), frozenset(e)]
+                                   for e in self.edge_iterator(labels=None)), min=1)
+                       
+                
+        # condition for induced subgraph ensures that if there
+        # doesnt exist an edge(h1, h2) in H then there should 
+        # not be an edge between representative sets of h1 and h2 in G
+        for h1 in H:
```

Comment:
This is simpler
```py
for h1, h2 in H.complement().edge_iterator(labels=False):
    for v1, v2 in G.self.edge_iterator(labels=False):
        p.add_constraint(rs[h1, v1] + rs[h2, v2], max=1)
        p.add_constraint(rs[h2, v1] + rs[h1, v2], max=1)
```

Reply:
Thanks, I have unified the code for minor and induced minor